aptos-move/aptos-vm/src/transaction_validation.rs

```diff
@@ -91,16 +101,23 @@ pub(crate) fn run_script_prologue(
     traversal_context: &mut TraversalContext,
     is_simulation: bool,
 ) -> Result<(), VMStatus> {
-    let txn_sequence_number = txn_data
-        .replay_protector()
-        .get_sequence_number()
-        .unwrap_or(0);
+    let txn_replay_protector = txn_data.replay_protector();
     let txn_authentication_key = txn_data.authentication_proof().optional_auth_key();
     let txn_gas_price = txn_data.gas_unit_price();
     let txn_max_gas_units = txn_data.max_gas_amount();
     let txn_expiration_timestamp_secs = txn_data.expiration_timestamp_secs();
     let chain_id = txn_data.chain_id();
     let mut gas_meter = UnmeteredGasMeter;
+
+    if !(features.is_transaction_payload_v2_enabled() && features.is_orderless_txns_enabled()) {
```

Comment:
can this be gated in move?
you just generate `replay_protector_move_value` and pass that to the move function and check the feature gating and abort in the move instead.
The current gating cannot be pruned later.

Reply:
It's sort of an empty assertion here because aptos_vm is already discarding these transactions when the flags are disabled. Removing this assertion here.

Reply:
I'm not sure if I can move the gating completely from aptos_vm to move though. Because the binary upgrade first, and framework upgrade happens later. Before the framework upgrade, aptos_vm won't be able to construct `replay_protector_move_value` and pass it to new prologue methods.

Reply:
I guess those should be put under `validate_signed_transaction`s together with keyless, etc

Reply:
Yes, those checks are currently in ** validate_signed_transaction**